main.py

```diff
@@ -89,7 +89,11 @@ def setup_target_task_training(args, target_tasks, model, strict):
     """
     Saves model states from pretraining if applicable, and
     loads the correct model state for the target task training
-    stage.
+    stage. 
+    The logic for loading the correct model state for target task training is:
+    1) If load_target_train_checkpoint is used, then load the weights from that checkpoint.
+    2) If we did pretraining, then load the best model from pretraining. 
+    3) Default case: We save untrained encoder weights.
 
     Parameters
     ----------------
```

Comment:
We should cut task_names_to_avoid_loading here, and generally scrap most of the existing code in this method.

Reply:
Actually, I realize that we should still keep task_names_to_avoid_loading since we want to avoid loading the task-specific modules before we finetune it.